### PR TITLE
WIP: Use ansible

### PR DESCRIPTION
**Pull request by bcrochet**
Before I go any further, what say y'all? Helpful or nah?

### CODE REVIEW

File: ansible.cfg [hunk 1]
```diff
@@ -0,0 +1,496 @@
+# config file for ansible -- https://ansible.com/
+# ===============================================
```
Comment (russellb) [inline ansible.cfg]: is it worth keeping all this boilerplate in tree?  I imagine it evovles with each version of ansible, so might as well just refer to ansible docs?

Comment (bcrochet) [reply ansible.cfg]: Meant to remove that. :)

**Comment by russellb** [comment]
I haven't looked at it in a ton of detail, but +1 to the concept from me!

**Comment by imain** [comment]
Yeah I dunno man.. :)

This whole thing basically evolved from a google doc and then just became an easier way to set it all up.  I consider this to be half documentation as well as a script that sets up the things we need.  IMO the ansible version looses that simplicity.  I like being able to look and see exactly what commands are needed to run in the order they are needed.  I don't see us needing to bring an orchestration tool into something which is just a simple set up of a node.

I'm not hard opposed to it mind you, but those are my thoughts.  I didn't even want Flavio to create a function for adding pull requests.. :)  to give you some idea where I'm at with this.  I like that a new person can come on board and just see what is happening without having to trace through dependency chains and variable expansions etc.

**Comment by hardys** [comment]
I think there are pros/cons - yes Ian is right it's slightly harder to follow the process with ansible (but given the nature of the simple playbooks e.g no roles or includes etc it's not bad IMO), but we gain idempotency for free because e.g the os_* modules know if we've already created openstack resources etc.

So I'm happy to go with the consensus - I like ansible and I'm fine with doing this if others agree, but equally I don't think it's 100% required here and shell scripts are fine for simple stuff like this too :)

**Comment by bcrochet** [comment]
> I think there are pros/cons - yes Ian is right it's slightly harder to follow the process with ansible (but given the nature of the simple playbooks e.g no roles or includes etc it's not bad IMO), but we gain idempotency for free because e.g the os_* modules know if we've already created openstack resources etc.

The idempotency was really my main driver. I find it easier to follow the playbook than the scripts, since there are fewer 'if-then-else' checks for things. 

Plus, it is basically self-documenting, given that each task has a descriptive name telling you what it's doing. :)

> 
> So I'm happy to go with the consensus - I like ansible and I'm fine with doing this if others agree, but equally I don't think it's 100% required here and shell scripts are fine for simple stuff like this too :)

I get not 100% required. I'd like to get a consensus on this before going any further though. Just an hour or two invested so far, so nothing lost if we decide to ditch it.

**Comment by bcrochet** [comment]
> Yeah I dunno man.. :)
> 
> This whole thing basically evolved from a google doc and then just became an easier way to set it all up. I consider this to be half documentation as well as a script that sets up the things we need. IMO the ansible version looses that simplicity. I like being able to look and see exactly what commands are needed to run in the order they are needed. I don't see us needing to bring an orchestration tool into something which is just a simple set up of a node.
> 
> I'm not hard opposed to it mind you, but those are my thoughts. I didn't even want Flavio to create a function for adding pull requests.. :) to give you some idea where I'm at with this. I like that a new person can come on board and just see what is happening without having to trace through dependency chains and variable expansions etc.

There aren't any dependency chains. I've done that on purpose. Just straight up Ansible. No roles, not even includes. and the variables are straight from the shell environment.

**Comment by dhellmann** [comment]
+1 LGTM

**Pull request closed without merging**